engine/docker/docker.go

```diff
@@ -212,6 +224,12 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 		logrus.WithField("network", pipelineConfig.Network.ID).WithField("error", err).Warnln("failed to remove network")
 	}
 
+	pruneFilters := filters.NewArgs()
+	pruneFilters.Add("label", harness)
+	if _, err := e.client.NetworksPrune(ctx, pruneFilters); err != nil {
```

Comment:
Will this interfere with parallel stages ? Both stages will have same label no ?

Reply:
prune only deletes the networks which are not in use, right?

Reply:
I see that there can be edge cases. Nice catch. Will change it.